src/main/kotlin/interactor/GetLowestAveragePricesForFruitsAndVegetablesInteractor.kt

```diff
@@ -1,40 +1,47 @@
 package interactor
 
 import model.CityEntity
+import utils.isNotNull
 
 class GetLowestAveragePricesForFruitsAndVegetablesInteractor (
     private val data: CostOfLivingDataSource
 ){
     fun execute(limit: Int): List<String> {
+
+        if (limit < 0) throw InvalidLimitException("Limit cannot be negative")
+
         return  data
             .getAllCitiesData()
             .filter(::excludeNullSalariesAndLowQualityDataAndNullFruitsAndVegetablesPrices)
-            .sortedBy {getRatio(it)}
-            .map { it.cityName }
+            .ifEmpty { throw IllegalStateException("Something went wrong, no valid data") }
+            .sortedBy(::getPricesToSalaryAverage)
             .take(limit)
+            .map { it.cityName }
```

Comment:
Remove **ifEmpty**, if the filtered data is empty(that means all data is filtered), then it returns an empty list.
Add a test case if the filtered list is empty then should return empty list.
```suggestion
            .sortedBy(::getPricesToSalaryAverage)
            .take(limit)
            .map { it.cityName }
```